core/frontend/src/tile/TileAdmin.ts

```diff
@@ -54,189 +54,475 @@ export interface SelectedAndReadyTiles {
  * a set of active requests. On each update it identifies previously-requested tiles whose content no viewport is interested in any longer and
  * cancels them. It then pulls pending requests off the queue and dispatches them into the active set until either the maximum number of
  * simultaneously-active requests is reached or the queue becomes empty.
- * @alpha
+ * @beta
  */
-export abstract class TileAdmin {
+export class TileAdmin {
+  private readonly _viewports = new Set<Viewport>();
+  private readonly _requestsPerViewport = new Map<Viewport, Set<Tile>>();
+  private readonly _tileUsagePerViewport = new Map<Viewport, Set<TileUsageMarker>>();
+  private readonly _selectedAndReady = new Map<Viewport, SelectedAndReadyTiles>();
+  private readonly _viewportSetsForRequests = new UniqueViewportSets();
+  private _maxActiveRequests: number;
+  private readonly _maxActiveTileTreePropsRequests: number;
+  private _defaultTileSizeModifier: number;
+  private readonly _retryInterval: number;
+  private readonly _enableInstancing: boolean;
+  private readonly _enableImprovedElision: boolean;
+  private readonly _ignoreAreaPatterns: boolean;
+  private readonly _enableExternalTextures: boolean;

```

Comment:
@markschlosseratbentley please double-check I didn't mess something up when merging your external textures changes into TileAdmin. (Your integration still passes so probably ok).

Reply:
@pmconne I went over your code, and it looks good with relation to external textures.  I also built + ran.  External textures all seem in order & working.